src/arch/aarch64/kernel/mod.rs

```diff
@@ -1,37 +1,37 @@
-// Copyright (c) 2018 Stefan Lankes, RWTH Aachen University
-//                    Colin Finck, RWTH Aachen University
+// Copyright (c) 2018-2020 Stefan Lankes, RWTH Aachen University
+//               2018 Colin Finck, RWTH Aachen University
 //
 // Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
 // http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
 // http://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
-include!(concat!(env!("CARGO_TARGET_DIR"), "/config.rs"));
-
 pub mod irq;
+pub mod pci;
 pub mod percore;
 pub mod processor;
 pub mod scheduler;
 pub mod serial;
+mod start;
 pub mod stubs;
+pub mod switch;
 pub mod systemtime;
 
 use crate::arch::aarch64::kernel::percore::*;
 use crate::arch::aarch64::kernel::serial::SerialPort;
 pub use crate::arch::aarch64::kernel::stubs::*;
 pub use crate::arch::aarch64::kernel::systemtime::get_boot_time;
+use crate::arch::aarch64::mm::{PhysAddr, VirtAddr};
+use crate::config::*;
 use crate::environment;
 use crate::kernel_message_buffer;
 use crate::synch::spinlock::Spinlock;
 use core::ptr;
 
 const SERIAL_PORT_BAUDRATE: u32 = 115200;
 
-lazy_static! {
-	static ref COM1: SerialPort = SerialPort::new(unsafe { BOOT_INFO.uartport });
-	static ref CPU_ONLINE: Spinlock<&'static mut u32> =
-		Spinlock::new(unsafe { &mut BOOT_INFO.cpu_online });
-}
+static mut COM1: SerialPort = SerialPort::new(0x9000000);
+static CPU_ONLINE: Spinlock<u32> = Spinlock::new(0);
 
 #[repr(C)]
 struct BootInfo {
```

Comment:
Why do we have one `struct BootInfo` here, but this MR also adds another one in src/arch/aarch64/kernel/bootinfo.rs?

Reply:
You are right, I removed one implementation